controller/backup_target_controller.go

```diff
@@ -329,25 +371,46 @@ func (btc *BackupTargetController) reconcile(name string) (err error) {
 		if reflect.DeepEqual(existingBackupTarget.Status, backupTarget.Status) {
 			return
 		}
-		if _, err := btc.ds.UpdateBackupTargetStatus(backupTarget); err != nil && apierrors.IsConflict(errors.Cause(err)) {
+		updatedBackupTarget, err := btc.ds.UpdateBackupTargetStatus(backupTarget)
+		if err != nil {
+			if apierrors.IsConflict(errors.Cause(err)) {
+				log.WithError(err).Debugf("Requeue %v due to conflict", name)
+				btc.enqueueBackupTarget(backupTarget)
+			}
+			if !apierrors.IsNotFound(err) {
+				log.WithError(err).Debugf("Failed to updated backup target %v", name)
+				return
+			}
+		}
+		if !backupTarget.DeletionTimestamp.IsZero() || updatedBackupTarget.Spec.Default == backupTarget.Spec.Default {
```

Comment:
I think we can move the deletion handling  [L396](https://github.com/longhorn/longhorn-manager/pull/2182/files#diff-b881cb9d9886c8b07c3081dc101c18685f066469dee34f80bcdb7b98f76acacdR396) to the above
So we don't need to consider `DeletionTimestamp` here and the following logic

Reply:
Modified and moved this logic before setting the timer.
Thanks.